addon/components/frost-link.js

```diff
@@ -138,10 +163,13 @@ export default LinkComponent.extend(PropTypeMixin, {
   init () {
     this._super(...arguments)
 
-    // Turn off the default _invoke on event and use _clickAndInvoke instead
-    let eventName = get(this, 'eventName')
-    this.off(eventName)
-    this.on(eventName, this, this._clickAndInvoke)
+    this._setTarget()
+  },
+
+  click () {
```

Comment:
I was wondering about that as well. I noticed that we are using the `EventProxy` for `frost-password`. I will open an issue to track moving to `EventProxy` for all of the components that it makes sense to do so.